src/server/Agent.js

```diff
@@ -72,36 +82,23 @@ export default class Agent extends SideAgent {
     // Take care of the given accepted connection.
     // The connection may have been accepted by another Agent instance.
     _startServing(userSocket) {
-        let transaction = null;
-        if (this._transaction.started()) {
-            transaction = new Transaction(this);
-
-            // reset() copies everything, so this cannot work well if
-            // the original response has transaction-specific pieces
-            assert(!this._originalResponse.finalized());
-            transaction.response.reset(this._originalResponse);
-        } else {
-            transaction = this._transaction;
-
-            // see the reset(this._originalResponse) comments above
-            assert(!transaction.response.finalized());
-            assert(!this._originalResponse);
-            this._originalResponse = transaction.response.clone();
-        }
-
-        return this._runTransaction(transaction, userSocket);
+        assert(!this._transaction.started());
+        assert(!this._transaction.response.finalized());
+        return this._runTransaction(this._transaction, userSocket);
     }
 
     async _stop() {
         if (this.server && this.server.address()) {
-            const promiseToClose = this.server.closeAsync(); // unconditionally
+            if (!this._serverClosed) // nil promise means that server has not accepted a connection
```

Comment:
Just to simplify the logic:

```suggestion
            if (!this._serverClosed) // we have not called this.server.closeAsync() yet
```

Removing `this._serverClosed = null` in another change request would help with this simplification.